server/src/main/java/org/opensearch/cluster/routing/allocation/AwarenessReplicaBalance.java

```diff
@@ -101,12 +102,22 @@ public int maxAwarenessAttributes() {
         return awarenessAttributes;
     }
 
-    public Optional<String> validate(int replicaCount) {
-        if ((replicaCount + 1) % maxAwarenessAttributes() != 0) {
-            String errorMessage = "expected total copies needs to be a multiple of total awareness attributes ["
-                + maxAwarenessAttributes()
-                + "]";
-            return Optional.of(errorMessage);
+    public Optional<String> validate(int replicaCount, AutoExpandReplicas autoExpandReplica) {
+        if (autoExpandReplica.isEnabled()) {
+            if ((autoExpandReplica.getMaxReplicas() != Integer.MAX_VALUE)
+                && ((autoExpandReplica.getMaxReplicas() + 1) % maxAwarenessAttributes() != 0)) {
```

Comment:
How is 0-N handled?

Reply:
if we use `0-all`, we get `Integer.MAX_VALUE` in max replicas